internal/state.go

```diff
@@ -65,18 +66,28 @@ func NewState(workerPool *WorkerPool, asg *types.AutoScalingGroup) (*State, erro
 		ASG:                  asg,
 		inServiceInstanceIDs: inServiceInstanceIDs,
 		workersByInstanceID:  workersByInstanceID,
+		cfg:                  cfg,
 	}, nil
 }
 
-// IdleWorkers returns a list of workers that are not currently busy.
-func (s *State) IdleWorkers() []Worker {
+// ScalableWorkers returns a list of workers that are not currently busy.
+func (s *State) ScalableWorkers() []Worker {
 	var out []Worker
-
 	for _, worker := range s.WorkerPool.Workers {
 		if worker.Busy {
 			continue
 		}
 
+		// Even though the worker might be idle, we will give it some time to pick up more work
+		// if the customer wants
+		if s.cfg.SpaceliftScaleDownDelay != 0 {
+			workerCreationTime := time.Unix(int64(worker.CreatedAt), 0)
+			minimumAliveTime := workerCreationTime.Add(time.Duration(s.cfg.SpaceliftScaleDownDelay) * time.Minute)
+			if !time.Now().After(minimumAliveTime) {
+				continue
+			}
+		}
```

Comment:
This is evaluated _per worker_, right? So if, say, a scale-down event would terminate 5 workers, but only 2 are too-new, the remaining 3 will still be terminated, right? Just want to make sure that the entire operation isn't cancelled if only a subset of workers don't meet the delay criterion.

Reply:
Correct, it's just going to skip individual workers that haven't been registered to spacelift long enough. It will still terminate other workers if they meet the criteria to be terminated.